src/libos/src/vm/vm_manager.rs

```diff
@@ -45,6 +56,23 @@ impl VMInitializer {
                     *b = 0;
                 }
             }
+            VMInitializer::CopyOldAndReadNew {
+                old_range,
+                file,
+                offset,
+            } => {
+                let src_slice = unsafe { old_range.as_slice() };
```

Comment:
There is a possibility that the old_range has a non-readable memory permission, causing the subsequent `copy_from_slice` to panic.